src/main/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptor.java

```diff
@@ -0,0 +1,178 @@
+package ca.uhn.fhir.jpa.starter.interceptor;
+
+import java.security.GeneralSecurityException;
+import java.util.List;
+
+import org.hl7.fhir.r4.model.IdType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.google.common.base.Strings;
+
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.jpa.starter.AppProperties;
+import ca.uhn.fhir.jpa.starter.util.ApiKeyHelper;
+import ca.uhn.fhir.jpa.starter.util.OAuth2Helper;
+import ca.uhn.fhir.rest.api.Constants;
+import ca.uhn.fhir.rest.api.RequestTypeEnum;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.exceptions.AuthenticationException;
+import ca.uhn.fhir.rest.server.interceptor.auth.AuthorizationInterceptor;
+import ca.uhn.fhir.rest.server.interceptor.auth.IAuthRule;
+import ca.uhn.fhir.rest.server.interceptor.auth.RuleBuilder;
+
+@Interceptor
+public class CustomAuthorizationInterceptor extends AuthorizationInterceptor {
+	private static final Logger logger = LoggerFactory.getLogger(CustomAuthorizationInterceptor.class);
+
+	private final AppProperties config;
+
+	public CustomAuthorizationInterceptor(AppProperties config) {
+		super();
+		this.config = config;
+	}
+
+	@Override
+	public List<IAuthRule> buildRuleList(RequestDetails theRequest) {
+		if (!isAnyEnabled()) {
+			return authorizedRule();
+		}
+
+		try {
+			if (Constants.URL_TOKEN_METADATA.equals(theRequest.getRequestPath())) {
+				return authorizedRule();
```

Comment:
function unauthorizedRule() is only allowing metadata, so instead of allowing all can we use the functionality same as unauthorizedRule function ?

Reply:
I don't think it matters which rule is returned here, but that's reasonable. I'll make the change.